### ISSUE TITLE
error checking version of psycopg2 

### ISSUE THREAD

**Issue by a01810** [zalando/patroni#1918]
Hello all.

After upgrade python3 from 3.6 to 3.8
```
~# systemctl start patroni
~# journalctl -xeu patroni

-- The start-up result is RESULT.
Apr 23 09:13:20 db02t patroni[21480]: FATAL: **Patroni requires psycopg2>=2.5.4 or psycopg2-binary**
Apr 23 09:13:20 db02t systemd[1]: patroni.service: Main process exited, code=exited, status=1/FAILURE
Apr 23 09:13:20 db02t systemd[1]: patroni.service: Failed with result 'exit-code'.
```
If I go back to python 3.6, then there is no problem with launching. (But it looks like there are problems with ssl-auth)

What could be the problem? (Except for crooked hands)) )

```
~# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.3 LTS
Release: 18.04
Codename:       bionic

~# python3 -V
Python 3.8.0

~#  dpkg -l | grep psyc
ii  python-psycopg2                        2.8.4-1~pgdg18.04+1                             amd64        Python module for PostgreSQL
ii  python3-psycopg2                       2.8.6-2~pgdg18.04+1                             amd64        Python 3 module for PostgreSQL


~# pip3 list | grep psyc
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
psycopg2 (2.8.6)

~# update-alternatives --config python3
There is 1 choice for the alternative python3 (providing /usr/bin/python3).

  Selection    Path                Priority   Status
------------------------------------------------------------
  0            /usr/bin/python3.8   10        auto mode
* 1            /usr/bin/python3.8   10        manual mode
```

Patroni version 2.0.2

**Comment by CyberDem0n** [comment]
> What could be the problem?

Which python version is Patroni starting with?
Have you reinstalled Patroni after upgrading python?

**Comment by a01810** [comment]
Patroni was installed with Python 3.6.
Not yet. I'm going to try now.

**Comment by a01810** [comment]
After apt purge - the same.

**Comment by a01810** [comment]
And patroni was updated from 1.14.?? (default in repo ubuntu 18.04) to 2.0.2. Not installing clean 2.0.2.

**Comment by CyberDem0n** [comment]
> After apt purge - the same.

Well, I can only recommend following the whole path:
1. check system unit and get the path to patroni
2. Open the patroni script that you got from 1 and check which python version is used
3. and so on.

**Comment by CyberDem0n** [comment]
```
~# cat /usr/bin/patroni
#!/usr/bin/python3
# EASY-INSTALL-ENTRY-SCRIPT: 'patroni==2.0.2','console_scripts','patroni'
[--cut--]
```

Perfect, now you have to find where Patroni is installed on the filesystem. It could be somewhere in the /usr/lib or /usr/local/lib.



>Is this the same issue #1021? Not fixed?

There is no issue with Patroni in this regard.
Patroni needs `psycopg2`, and in case if it can't be loaded (or very outdated) it complains and exits.
On your system Patroni can't find psycopg2.

**Comment by a01810** [comment]
```
~# ll /usr/lib/python3/dist-packages/patroni
total 304
drwxr-xr-x   7 root root  4096 Apr 23 11:47 ./
drwxr-xr-x 130 root root 12288 Apr 23 11:47 ../
-rw-r--r--   1 root root 30786 Feb 22 14:28 api.py
-rw-r--r--   1 root root  4661 Feb 22 14:28 async_executor.py
-rw-r--r--   1 root root 18811 Feb 22 14:28 config.py
-rw-r--r--   1 root root 52143 Feb 22 14:28 ctl.py
[---cut---]
```

```
~# ll /usr/lib/python3/dist-packages/psycopg2
total 436
drwxr-xr-x   3 root root   4096 Apr 21 19:08 ./
drwxr-xr-x 130 root root  12288 Apr 23 11:47 ../
-rw-r--r--   1 root root    367 Sep  6  2020 compat.py
-rw-r--r--   1 root root  14273 Sep  6  2020 errorcodes.py
-rw-r--r--   1 root root   1420 Sep  6  2020 errors.py
-rw-r--r--   1 root root   7122 Sep  6  2020 extensions.py
[---cut---]
```
They are near...

**Comment by CyberDem0n** [comment]
Patroni perfectly works with python3.8 and on Ubuntu 20.04. For every commit there are plenty of tests executed https://github.com/zalando/patroni/runs/2388829097?check_suite_focus=true and as you can see they all succeed.

Patroni is not doing any magic, it only trying to `import psycopg2`, which fails. I can only suggest starting from scratch. Or maybe use `virtualenv`.

**Comment by a01810** [comment]
"In 20.04 LTS, the python included in the base system is Python 3.8."
18.04 - Python 3.6. 
Ok. I'll check.
